src/frontend/hooks/useSendMetrics.ts

```diff
@@ -0,0 +1,129 @@
+  assert(result, `${name} environment variable should be set`);
+  return result;
+};
+
+type LastMetricsReportSentAtSlice = {
+  lastMetricsReportSentAt: null | number;
+  setLastMetricsReportSentAt: (lastMetricsReportSentAt: number) => void;
+};
+
+const lastMetricsReportSentAtSlice: StateCreator<
+  LastMetricsReportSentAtSlice
+> = set => ({
+  lastMetricsReportSentAt: null,
+  setLastMetricsReportSentAt: lastMetricsReportSentAt =>
+    set({lastMetricsReportSentAt}),
+});
+
+const useLastMetricsReportSentAt = createPersistedState(
+  lastMetricsReportSentAtSlice,
+  'LastMetricsReportSentAt',
+);
+
+/**
+ * Sleep until we can send metrics again.
+ *
+ * Throws if aborted.
+ */
+const sleepUntilWeCanSendMetrics = (
+  lastMetricsReportSentAt: null | number,
+  options: Readonly<{signal: AbortSignal}>,
+): Promise<void> => {
+  if (lastMetricsReportSentAt == null) lastMetricsReportSentAt = -Infinity;
+  const earliestTimeNextMetricsReportCouldBeSentAt =
+    lastMetricsReportSentAt + SEND_METRICS_INTERVAL;
+  const waitTime = Math.max(
+    0,
+    earliestTimeNextMetricsReportCouldBeSentAt - Date.now(),
+  );
+  return sleep(waitTime, options);
+};
+
+const getAllObservations = async (
+  api: MapeoClientApi,
+): Promise<ObservationValue[]> => {
+  const projectSettings = await api.listProjects();
+  return (
+    await Promise.all(
+      projectSettings.map(async ({projectId}) => {
+        const project = await api.getProject(projectId);
+        return project.observation.getMany();
+      }),
+    )
+  ).flat();
+};
+
+export default function useSendMetrics(api: MapeoClientApi): void {
+  const lastMetricsReportSentAt = useLastMetricsReportSentAt(
+    store => store.lastMetricsReportSentAt,
+  );
+  const setLastMetricsReportSentAt = useLastMetricsReportSentAt(
+    store => store.setLastMetricsReportSentAt,
+  );
+
+  const isAppActive = useAppState() === 'active';
+
+  useEffect(() => {
+    if (!isAppActive) return;
+
+    const abortController = new AbortController();
+    const {signal} = abortController;
+
+    (async () => {
+      const metricsUrl = getEnv('EXPO_PUBLIC_METRICS_URL');
+      const metricsApiKey = getEnv('EXPO_PUBLIC_METRICS_API_KEY');
+
+      await sleepUntilWeCanSendMetrics(lastMetricsReportSentAt, {signal});
+
+      const metricsReport = generateMetricsReport({
+        packageJson,
+        os: Platform.OS,
+        osVersion: Platform.Version,
+        screen: Dimensions.get('screen'),
+        observations: await getAllObservations(api),
+      });
+
+      throwIfAborted(signal);
+
+      // We do everything other than actually send the request in dev to be as
+      // realistic as possible without sending bogus data to the server.
+      if (!__DEV__) {
+        await sendMetricsReport({
+          metricsUrl,
+          metricsApiKey,
+          metricsReport,
+          signal,
+        });
+      }
+
+      setLastMetricsReportSentAt(Date.now());
+    })().catch(noop);
```

Comment:
Instead of `noop', we should pass the errors to Sentry (at least the server errors, but probably also the `getenv` errors)

Reply:
Done in e7b3f922d1267ef2c6f7b2b32a5a053c711d6a8c.